cassandra-mesos-scheduler/src/main/java/io/mesosphere/mesos/frameworks/cassandra/scheduler/SeedManager.java

```diff
@@ -0,0 +1,110 @@
+package io.mesosphere.mesos.frameworks.cassandra.scheduler;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.google.common.base.Joiner;
+import org.jetbrains.annotations.NotNull;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.net.HttpURLConnection;
+import java.net.URL;
+import java.util.List;
+import java.util.concurrent.TimeUnit;
+
+import static com.google.common.collect.Lists.newArrayList;
+import static io.mesosphere.mesos.frameworks.cassandra.CassandraFrameworkProtos.CassandraFrameworkConfiguration;
+import static io.mesosphere.mesos.frameworks.cassandra.CassandraFrameworkProtos.ExternalDc;
+
+public class SeedManager {
+    public static final long FETCH_PERIOD = TimeUnit.MINUTES.toMillis(1);
+    private static final Logger LOGGER = LoggerFactory.getLogger(SeedManager.class);
+
+    @NotNull
+    private final PersistedCassandraFrameworkConfiguration configuration;
+
+    public SeedManager(@NotNull PersistedCassandraFrameworkConfiguration configuration) {
+        this.configuration = configuration;
+    }
+
+    @NotNull
+    public List<String> getSeeds() {
+        final List<String> seeds = newArrayList();
+
+        for (final ExternalDc dc : configuration.get().getExternalDcsList()) {
+            seeds.addAll(dc.getSeedsList());
+        }
+
+        return seeds;
+    }
+
+    public void syncSeeds(long now) {
+        for (ExternalDc dc : configuration.get().getExternalDcsList()) {
+            fetchSeedsIfRequired(dc, now);
+        }
+    }
+
+    private void fetchSeedsIfRequired(ExternalDc dc, long now) {
```

Comment:
Remove `now` parameter, and replace with `Clock` once it's passed into this class.